### PR TITLE
use prepared statement for sqlite insert

### PR DESCRIPTION
**Pull request by avieth**
Expected to give at least a marginal performance boost.

### CODE REVIEW

File: src/lib/Ouroboros/Byron/Proxy/Index/Sqlite.hs [hunk 1]
```diff
@@ -42,13 +42,16 @@ data TraceEvent where
...
+  :: EpochSlots
+  -> Tracer IO TraceEvent
+  -> Connection
+  -> Sql.Statement -- for insert
+  -> Index IO (Header (Block cfg))
+index epochSlots tracer conn insertStatement = Index
   { Index.lookup = sqliteLookup epochSlots conn
   , tip          = sqliteTip epochSlots conn
-  , rollforward  = sqliteRollforward epochSlots tracer conn
+  , rollforward  = sqliteRollforward epochSlots tracer insertStatement
```
Comment (cleverca22) [inline src/lib/Ouroboros/Byron/Proxy/Index/Sqlite.hs:54]: something i was thinking of, is to replace `Connection` with a data record, that holds the `Connection` and each `Statement` object

then you can pass that same object around and not have to rewrite the whole call stack every time you add a new prepared statement

Comment (avieth) [reply src/lib/Ouroboros/Byron/Proxy/Index/Sqlite.hs:54]: Will do that if I ever add another statement. Currently there's just 1 so it's no big deal.

**Review by coot — APPROVED** [review]
LGTM

**Pull request merged**